cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2383,6 +2394,35 @@ int getPodIndexFromPvcName(String pvcName)  {
             }
         }
 
+        Future<ReconciliationState> entityOperatorReady() {
+            long pollingIntervalMs = 1_000;
+            long timeoutMs = operationTimeoutMs;
+
+            if (this.entityOperator != null && eoDeployment != null) {
+                Future<Void> future = podOperations.readiness(namespace, this.entityOperator.getName(), 1_000, timeoutMs);
+                future.compose(v -> {
+                    Condition condition;
```

Comment:
I do not think we necessarily need a separate condition for the EO. We just need to wait until it is ready and then set the main condition to ready.